crates/re_query_cache/src/cache.rs

```diff
@@ -120,13 +153,145 @@ impl CacheKey {
         store_id: impl Into<StoreId>,
         entity_path: impl Into<EntityPath>,
         timeline: impl Into<Timeline>,
-        archetype_name: impl Into<ArchetypeName>,
     ) -> Self {
         Self {
             store_id: store_id.into(),
             entity_path: entity_path.into(),
             timeline: timeline.into(),
-            archetype_name: archetype_name.into(),
+        }
+    }
+}
+
+// --- Invalidation ---
+
+impl StoreSubscriber for Caches {
+    #[inline]
+    fn name(&self) -> String {
+        "rerun.store_subscribers.QueryCache".into()
+    }
+
+    #[inline]
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    #[inline]
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
+        self
+    }
+
+    // TODO(cmc): support dropped recordings.
```

Comment:
Didn't you start working on this at some point @jleibs? Or was it @emilk? My memory is all fuzzy...